src/main/java/org/gridsuite/network/map/NetworkMapService.java

```diff
@@ -864,6 +872,19 @@ public List<VoltageLevelMapData> getVoltageLevels(UUID networkUuid, String varia
             substationsId.stream().flatMap(id -> network.getSubstation(id).getVoltageLevelStream().map(NetworkMapService::toMapData)).collect(Collectors.toList());
     }
 
+    public List<VoltageLevelsEquipmentsMapData> getVoltageLevelsAndConnectable(UUID networkUuid, String variantId, List<String> substationsId) {
+        Network network = getNetwork(networkUuid, substationsId == null ? PreloadingStrategy.COLLECTION : PreloadingStrategy.NONE, variantId);
+        List<VoltageLevel> voltageLevels =  substationsId == null ?
```

Comment:
You could have kept the "Stream style" all the way, as in `getVoltageLevels` above
```
    public List<VoltageLevelsEquipmentsMapData> getVoltageLevelsAndConnectable(UUID networkUuid, String variantId, List<String> substationsId) {
        Network network = getNetwork(networkUuid,
            substationsId == null ? PreloadingStrategy.COLLECTION : PreloadingStrategy.NONE,
            variantId);
        Stream<VoltageLevel> voltageLevels = substationsId == null ?
            network.getVoltageLevelStream() :
            substationsId.stream().flatMap(id -> network.getSubstation(id).getVoltageLevelStream());

        return voltageLevels.map(vl -> VoltageLevelsEquipmentsMapData.builder()
                .voltageLevel(toMapData(vl))
                .equipments(vl.getConnectableStream().map(NetworkMapService::toMapData).collect(Collectors.toList()))
                .build())
            .collect(Collectors.toList());
    }
```